File: app/src/main/java/com/mandarin/bcu/androidutil/stage/adapters/LimitRecycle.java
package com.mandarin.bcu.androidutil.stage.adapters;

import android.app.Activity;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.mandarin.bcu.R;
import com.mandarin.bcu.androidutil.getStrings;

import common.util.stage.Limit;

public class LimitRecycle extends RecyclerView.Adapter<LimitRecycle.ViewHolder>{
    private final Activity activity;
    private String [] limits;

    public LimitRecycle(Activity activity, Limit l) {
        this.activity = activity;

        getStrings s = new getStrings(activity);

        limits = s.getLimit(l);
    }

    class ViewHolder extends RecyclerView.ViewHolder {
        TextView limit;

        ViewHolder(@NonNull View itemView) {
            super(itemView);

            limit = itemView.findViewById(R.id.limitst);
        }
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        View row = LayoutInflater.from(activity).inflate(R.layout.stg_limit_layout,viewGroup,false);

        return new ViewHolder(row);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder viewHolder, int i) {
        viewHolder.limit.setText(limits[viewHolder.getAdapterPosition()]);
    }

    @Override
    public int getItemCount() {
        return limits.length;
    }
}

File: app/src/main/java/com/mandarin/bcu/androidutil/stage/adapters/LimitRecycle.kt
package com.mandarin.bcu.androidutil.stage.adapters

import android.app.Activity
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.mandarin.bcu.R
import com.mandarin.bcu.androidutil.GetStrings
import common.util.stage.Limit

class LimitRecycle(private val activity: Activity, l: Limit?) : RecyclerView.Adapter<LimitRecycle.ViewHolder>() {
    private val limits: Array<String>

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var limit: TextView = itemView.findViewById(R.id.limitst)

    }

    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): ViewHolder {
        val row = LayoutInflater.from(activity).inflate(R.layout.stg_limit_layout, viewGroup, false)
        return ViewHolder(row)
    }

    override fun onBindViewHolder(viewHolder: ViewHolder, i: Int) {
        viewHolder.limit.text = limits[viewHolder.adapterPosition]
    }

    override fun getItemCount(): Int {
        return limits.size
    }

    init {
        val s = GetStrings(activity)
        limits = s.getLimit(l)
    }
}